juspay/hyperswitch
# refactor(access_token): handle timeout errors gracefully

## Type of Change



## Description

If there is any request timeout error in any flow before making a payment at the connector, then those errors have to be handled gracefully. This PR introduces it for access token flow.


## Motivation and Context



## How did you test it?

- Create a payment by 
<img width="1295" alt="Screenshot 2023-08-07 at 1 13 15 PM" src="https://github.com/juspay/hyperswitch/assets/48803246/0a44fbb8-835b-4f1c-ae91-8706f6a00608">
payu, to ensure that the previous flows are not affected.

- Change the connector url to something which will always timeout, https://google.com:81 and the payment should fail.
<img width="1245" alt="Screenshot 2023-08-07 at 1 30 05 PM" src="https://github.com/juspay/hyperswitch/assets/48803246/c2df1df9-3d40-42bb-a823-1bcf3e845f36">


## Checklist
